.github/workflows/e2e.yaml

```diff
@@ -8,6 +8,8 @@ on:
       - main
       - v*
     paths:
+      - go.mod
```

Comment:
Dependency changes can accidentally break vcluster code, so we should run tests when we change dependencies.